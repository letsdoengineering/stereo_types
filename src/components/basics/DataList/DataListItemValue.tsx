import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import './DataList.css'

type DataListItemValueProps = React.AllHTMLAttributes<HTMLElement> & {
  /** optional: class-names */
  className?: string
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** DataListItemValue: renders a DD tag */
const DataListItemValue: React.FC<DataListItemValueProps> = ({
  className,
  children,
}: DataListItemValueProps) => {
  const classNames = getClassNames({
    defaultClasses: ['data_list_item-value'],
    className,
  })

  return <dd className={classNames}>{children}</dd>
}

export default DataListItemValue
