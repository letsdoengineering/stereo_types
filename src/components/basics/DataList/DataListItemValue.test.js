import * as React from 'react'
import { render, waitFor, screen } from '@testing-library/react'
import DataListItemValue from './DataListItemValue'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

describe('DataListItemValue', () => {
  it('renders li', async () => {
    render(<DataListItemValue>a value</DataListItemValue>)
    const list = screen.getByText('a value')

    await waitFor(() => expect(list.nodeName).toBe('DD'))
  })

  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<DataListItemValue>a value</DataListItemValue>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<DataListItemValue>a value</DataListItemValue>)
    /** dLitem rule ignored as this component on its own does not sit inside a <dl> */
    const results = await axe(container, {
      rules: {
        dlitem: { enabled: false },
      },
    })
    expect(results).toHaveNoViolations()
  })
})
