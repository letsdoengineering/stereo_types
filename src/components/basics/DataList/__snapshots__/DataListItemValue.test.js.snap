// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataListItemValue has not changed the JSX layout of the page from previous snapshot 1`] = `
<dd
  className="itemValue"
>
  a value
</dd>
`;
